contracts/src/v0.8/keystone/CapabilityRegistry.sol

```diff
@@ -183,19 +184,20 @@ contract CapabilityRegistry is OwnerIsCreator, TypeAndVersionInterface {
       ) revert InvalidCapabilityConfigurationContractInterface(capability.configurationContract);
     }
 
-    s_capabilityIds.add(capabilityId);
-    s_capabilities[capabilityId] = capability;
+    s_compressedCapabilityIds.add(compressedId);
+    s_capabilities[compressedId] = capability;
 
-    emit CapabilityAdded(capabilityId);
+    emit CapabilityAdded(compressedId);
   }
 
-  function getCapability(bytes32 capabilityID) public view returns (Capability memory) {
-    return s_capabilities[capabilityID];
+  /// @notice This function returns a Capability by its compressed ID. Use `getCompressedCapabilityID` to get the compressed ID.
+  function getCapability(bytes32 compressedId) public view returns (Capability memory) {
+    return s_capabilities[compressedId];
   }
 
   /// @notice This functions returns a Capability ID packed into a bytes32 for cheaper access
   /// @return bytes32 A unique identifier for the capability
-  function getCapabilityID(bytes32 capabilityType, bytes32 version) public pure returns (bytes32) {
-    return keccak256(abi.encodePacked(capabilityType, version));
+  function getCompressedCapabilityID(bytes32 labeledName, bytes32 version) public pure returns (bytes32) {
```

Comment:
Is it really a reduction from 64 bytes down to 32? If we are so close then maybe it's just possible to require all IDs to fit in 32 bytes? I think that could simplify things quite a bit. Can we try to come up with some reasonable limits on labels or whatever takes most space?

Reply:
> Is it really a reduction from 64 bytes down to 32?
We take in two values with bytes32 and return bytes32, so it is a reduction to 32 bytes.

> If we are so close then maybe it's just possible to require all IDs to fit in 32 bytes?
I don't think that's realistic. Assuming the simplest version `XX.XX.XX` it already takes up 8 bytes. `read_chain-ethereum:mainnet` would not fit into the 24 remaining bytes (it is 27 bytes long). 32 bytes for capability type is restrictive as it is. If anything, we might want to consider expanding it to allow things like scope, e.g., `@chainlink` prefix.

Reply:
That's a bit sad but OK, if we need more then let's use a compressed name separately.